mobile/lib/modules/shared_link/providers/shared_link.provider.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/shared_link/services/shared_link.service.dart';
+import 'package:openapi/api.dart';
```

Comment:
could you rework this (and the service class) so that openapi (and thus the DTOs) are only used in the service - but not in the provider? This makes the app code more independent  of the DTOs (which might change any time without someone looking at the mobile code)

Reply:
I've updated the provider and service as requested.